pkg/iac-providers/terraform/v12/variable-references.go

```diff
@@ -148,6 +148,10 @@ func (r *RefResolver) ResolveVarRefFromParentModuleCall(varRef string) interface
 	if reflect.TypeOf(val).Kind() == reflect.String {
 		valStr := val.(string)
 		resolvedVal := strings.Replace(varRef, varExpr, valStr, 1)
+		if varRef == resolvedVal {
+			zap.S().Debugf("resolved str variable ref refers to self: '%v'", varRef)
+			return val
```

Comment:
It would be better to return `varRef` from here rather than returning `val`

Reply:
Fixed in https://github.com/accurics/terrascan/pull/393/commits/38d98144450724c13eb667220d9decbe4e9bd617